modules/common/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/utils/IgluUtils.scala

```diff
@@ -243,37 +216,45 @@ object IgluUtils {
       .leftMap((sdj.schema, _))
   }
 
-  /** Check a list of SDJs and merge the Iglu errors */
+  /**
+   * Check a list of SDJs.
+   * @return All valid SDJs are in the Right while all errors are in the Left
+   */
   private def checkList[F[_]: Monad: Clock](
     client: IgluCirceClient[F],
     sdjs: List[SelfDescribingData[Json]],
     registryLookup: RegistryLookup[F]
-  ): EitherT[F, NonEmptyList[(SchemaKey, ClientError)], Unit] =
-    EitherT {
-      sdjs
-        .map(check(client, _, registryLookup).toValidatedNel)
-        .sequence
-        .map(_.sequence_.toEither)
-    }
+  ): IorT[F, NonEmptyList[(SchemaKey, ClientError)], List[SelfDescribingData[Json]]] =
+    sdjs
```

Comment:
Sharing this in case you like it:

```scala
    sdjs.map { sdj =>
      check(client, sdj, registryLookup)
        .map(_ => List(sdj))
        .leftMap(NonEmptyList.one)
        .toIor
        .recoverWith { case errors => IorT.fromIor[F](Ior.Both(errors, Nil)) }
    }.foldA
```

Reply:
Yes I like it, it's fewer letters :laughing: